Jenkinsfile.trigger

```
@@ -102,69 +102,98 @@ if (breakEarly) { return } // thanks Jenkins...
 
 // new data to be added to the past-jobs.json
 // { lastTime: unixTimestamp, url: "" }
-def buildCompletionData = [:]
+buildCompletionData = [:]
 
-for (buildObj in queue) {
-	stage(buildObj.identifier) {
-		//def json = writeJSON(json: buildObj, returnText: true)
-		//echo(json) // for debugging/data purposes
+// list of closures that we can use to wait for the jobs on.
+def waitQueue = [:]
+def addToWait(identifier, buildId, externalizableId) {
```

I recall your original version of this function actually _was_ adding the given parameters to the queue, but given the complications with that IMO we should rename this to make it more clear it only returns the closure and adding it to the queue is still something the caller is responsible for; maybe something like this?

```suggestion
def waitQueueClosure(identifier, buildId, externalizableId) {
```